src/handle_vote.cxx

```diff
@@ -77,9 +77,11 @@ void raft_server::request_prevote() {
 
             if (recreate) {
                 p_in("reset RPC client for peer %d", s_config->get_id());
-                pp->recreate_rpc(s_config, *ctx_);
-                pp->set_free();
-                pp->set_manual_free();
+                bool ok = pp->recreate_rpc(s_config, *ctx_);
+                if (ok) {
+                    pp->set_free();
+                    pp->set_manual_free();
+                }
```

Comment:
Seems that every time `recreate_rpc()` call success, `set_free()` and `set_manual_free()` must be called.  Better call them inside `recreate_rpc()` before `return true`.

Reply:
Makes sense, thanks.